packages/features/bookings/lib/handleCancelBooking.ts

```diff
@@ -280,103 +281,43 @@ async function handler(req: NextApiRequest & { userId?: number }) {
 
   /** TODO: Remove this without breaking functionality */
   if (bookingToDelete.location === DailyLocationType) {
-    bookingToDelete.user.credentials.push(FAKE_DAILY_CREDENTIAL);
+    bookingToDelete.user[0].credentials.push(FAKE_DAILY_CREDENTIAL);
   }
 
   const apiDeletes = [];
 
-  const bookingCalendarReference = bookingToDelete.references.find((reference) =>
-    reference.type.includes("_calendar")
-  );
-
-  if (bookingCalendarReference) {
-    const { credentialId, uid, externalCalendarId } = bookingCalendarReference;
-    // If the booking calendar reference contains a credentialId
-    if (credentialId) {
-      // Find the correct calendar credential under user credentials
-      const calendarCredential = bookingToDelete.user.credentials.find(
-        (credential) => credential.id === credentialId
-      );
-      if (calendarCredential) {
-        const calendar = getCalendar(calendarCredential);
+  // Loop through booking references
+  bookingToDelete.references.forEach(async (reference) => {
+    if (reference.type.includes("_calendar")) {
+      if (reference.credential) {
+        const calendar = getCalendar(reference.credential);
+        const { uid, externalCalendarId } = reference;
```

Comment:
The following loop will go through each booking reference and read credentials linked to the reference. This is more efficient than combing through all user credentials. This also opens us up to improve the booking experience for teams (ex. updating multiple calendars)